File: app/src/main/java/com/itachi1706/cheesecakeutilities/Modules/MSLIntegration/CalendarInfo.java
package com.itachi1706.cheesecakeutilities.Modules.MSLIntegration;

import com.google.android.gms.common.internal.Objects;
import com.google.api.services.calendar.model.Calendar;
import com.google.api.services.calendar.model.CalendarListEntry;

/**
 * Created by Kenneth on 13/1/2019.
 * for com.itachi1706.cheesecakeutilities.Modules.MSLIntegration in CheesecakeUtilities
 */
class CalendarInfo implements Comparable<CalendarInfo>, Cloneable {

    static final String FIELDS = "id,summary";
    static final String FEED_FIELDS = "items(" + FIELDS + ")";

    String id;
    String summary;

    CalendarInfo(String id, String summary) {
        this.id = id;
        this.summary = summary;
    }

    CalendarInfo(Calendar calendar) {
        update(calendar);
    }

    CalendarInfo(CalendarListEntry calendar) {
        update(calendar);
    }

    @Override
    public String toString() {
        return Objects.toStringHelper(CalendarInfo.class).add("id", id).add("summary", summary)
                .toString();
    }

    public int compareTo(CalendarInfo other) {
        return summary.compareTo(other.summary);
    }

    @Override
    public CalendarInfo clone() {
        try {
            return (CalendarInfo) super.clone();
        } catch (CloneNotSupportedException exception) {
            // should not happen
            throw new RuntimeException(exception);
        }
    }

    void update(Calendar calendar) {
        id = calendar.getId();
        summary = calendar.getSummary();
    }

    void update(CalendarListEntry calendar) {
        id = calendar.getId();
        summary = calendar.getSummary();
    }
}

File: app/src/main/java/com/itachi1706/cheesecakeutilities/Modules/MSLIntegration/model/CalendarInfo.kt
package com.itachi1706.cheesecakeutilities.Modules.MSLIntegration.model

import com.google.android.gms.common.internal.Objects
import com.google.api.services.calendar.model.Calendar
import com.google.api.services.calendar.model.CalendarListEntry

/**
 * Created by Kenneth on 13/1/2019.
 * for com.itachi1706.cheesecakeutilities.Modules.MSLIntegration in CheesecakeUtilities
 */
@Deprecated("MSL Closing Down")
class CalendarInfo : Comparable<CalendarInfo>, Cloneable {

    private var id: String? = null
    private var summary: String? = null

    internal constructor(id: String, summary: String) {
        this.id = id
        this.summary = summary
    }

    internal constructor(calendar: Calendar) {
        update(calendar)
    }

    internal constructor(calendar: CalendarListEntry) {
        update(calendar)
    }

    override fun toString(): String {
        return Objects.toStringHelper(CalendarInfo::class.java).add("id", id).add("summary", summary)
                .toString()
    }

    override fun compareTo(other: CalendarInfo): Int {
        return summary!!.compareTo(other.summary!!)
    }

    public override fun clone(): CalendarInfo {
        try {
            return super.clone() as CalendarInfo
        } catch (exception: CloneNotSupportedException) {
            // should not happen
            throw RuntimeException(exception)
        }

    }

    fun update(calendar: Calendar) {
        id = calendar.id
        summary = calendar.summary
    }

    fun update(calendar: CalendarListEntry) {
        id = calendar.id
        summary = calendar.summary
    }

    companion object {

        const val FIELDS = "id,summary"
        const val FEED_FIELDS = "items($FIELDS)"
    }
}
